### PR TITLE
[vectorized](function) support time_to_sec function

### PR DESCRIPTION
**Pull request by zhangstar333**
# Proposed changes

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zhangstar333** [comment]
run buildall

File: docs/zh-CN/docs/sql-manual/sql-functions/date-time-functions/time_to_sec.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+## time_to_sec
+### description
+#### Syntax
+
+`INT time_to_sec(DATETIME datetime)`
```
Comment (HappenLee) [inline docs/zh-CN/docs/sql-manual/sql-functions/date-time-functions/time_to_sec.md:31]: We should support `TIME` type not `DATETIME` type like mysql

**Comment by zhangstar333** [comment]
run buildall

**Review by HappenLee — COMMENTED** [review]
Add regression test please

File: be/src/vec/functions/function_date_or_datetime_computation.h [hunk 2]
```diff
@@ -1001,6 +1002,24 @@ struct CurrentTimeImpl {
     }
 };
 
+struct TimeToSecImpl {
+    using ReturnType = DataTypeInt32;
+    static constexpr auto name = "time_to_sec";
+    static Status execute(FunctionContext* context, Block& block, const ColumnNumbers& arguments,
+                          size_t result, size_t input_rows_count) {
+        auto res_col = ColumnVector<Int32>::create();
+        auto argument_column =
```
Comment (HappenLee) [inline be/src/vec/functions/function_date_or_datetime_computation.h:1011]: do not use `convert_to_full_column_if_const`

File: be/src/vec/functions/function_date_or_datetime_computation.h [hunk 3]
```diff
@@ -1001,6 +1002,24 @@ struct CurrentTimeImpl {
...
+                block.get_by_position(arguments[0]).column->convert_to_full_column_if_const();
+        const auto& column_data = assert_cast<const ColumnFloat64&>(*argument_column);
+        for (int i = 0; i < input_rows_count; ++i) {
```
Comment (HappenLee) [inline be/src/vec/functions/function_date_or_datetime_computation.h:1014]: resize（）
res_col[i] = time

File: be/src/vec/functions/function_date_or_datetime_computation.h (hunk 2 above)
Comment (HappenLee) [inline be/src/vec/functions/function_date_or_datetime_computation.h:1011]: do not use `convert_to_full_column_if_const`

**Comment by zhangstar333** [comment]
can't create a column the type is time, if use current_time write regression test, the result is unstable

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run buildall

**Comment by zhangstar333** [comment]
run p0

**Comment by zhangstar333** [comment]
run p0

**Review by HappenLee — APPROVED** [review]
LGTM

**Pull request merged**